src/go/k8s/cli/configurator/main.go

```diff
@@ -0,0 +1,163 @@
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+// main package. This line should be remove after merging the following
+// https://github.com/vectorizedio/redpanda/pull/753
+package main
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/spf13/afero"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/config"
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	hostName		string
+	svcFQDN			string
+	configSourceDir		string
+	configDestination	string
+	redpandaRPCPort		int
+)
+
+var errorMissingEnvironmentVariable = errors.New("missing environment variable")
+
+func main() {
+	if err := checkEnvVars(); err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to get the environment variables: %w", err))
+	}
+
+	fs := afero.NewOsFs()
+	v := config.InitViper(fs)
+	v.AddConfigPath(configSourceDir)
+
+	if err := v.ReadInConfig(); err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to read the redpanda configuration file: %w", err))
+	}
+
+	cfg := &config.Config{}
+	decoderConfig := mapstructure.DecoderConfig{
+		Result:	cfg,
+		// Sometimes viper will save int values as strings (i.e.
+		// through BindPFlag) so we have to allow mapstructure
+		// to cast them.
+		WeaklyTypedInput:	true,
+	}
+
+	decoder, err := mapstructure.NewDecoder(&decoderConfig)
+	if err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to create decoder config: %w", err))
+	}
+
+	err = decoder.Decode(v.AllSettings())
+	if err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to decode: %w", err))
+	}
+
+	log.Printf("Decode done")
+
+	hostIndex, err := hostIndex(hostName)
+	if err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to extract host index: %w", err))
+	}
+
+	log.Printf("Host index calculated %d", hostIndex)
+
+	cfg.Redpanda.Id = hostIndex
+	if hostIndex == 0 {
+		cfg.Redpanda.SeedServers = []config.SeedServer{}
+	} else {
+		cfg.Redpanda.SeedServers = []config.SeedServer{
+			{
+				Host: config.SocketAddress{
+					// Example address: cluster-sample-0.cluster-sample.default.svc.cluster.local
+					Address:	hostName + svcFQDN,
+					Port:		redpandaRPCPort,
+				},
+			},
+		}
+	}
+
+	cfgBytes, err := yaml.Marshal(cfg)
+	if err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to marshal the configuration: %w", err))
+	}
+
+	log.Printf("Config: %s", string(cfgBytes))
+
+	if err := ioutil.WriteFile(configDestination, cfgBytes, 0600); err != nil {
+		log.Fatalf("%s", fmt.Errorf("unable to write the destination configuration file: %w", err))
+	}
+
+	log.Printf("Configuration saved into: %s", configDestination)
+}
+
+func checkEnvVars() error {
+	log.Printf("The redpanda configurator is starting")
+
+	var exist bool
+	hostName, exist = os.LookupEnv("HOSTNAME")
```

Comment:
nit: would be nice to define all these as constants in the top.

Reply:
we should also perhaps check all envs and return all errors so if you don't provide two env vars, you'll see both in the error message

Reply:
Ok will do